src/main/java/aspects/LockAspect.java

```diff
@@ -84,8 +84,24 @@ private void logLockError(AuthenticatedRequestBean bean, ProceedingJoinPoint joi
         );
     }
 
-    private Lock getLockAndBlock(String username) throws LockError {
+    private Lock getLockAndBlock(String username, ProceedingJoinPoint joinPoint)
+            throws LockError {
+        long startTime = System.nanoTime();
         Lock lock = userLockRegistry.obtain(username);
+        long timeInMs = (System.nanoTime() - startTime) / 1000000;
+        raven.newBreadcrumb()
+                .setCategory(Constants.TimingCategories.WAIT_ON_LOCK)
+                .setMessage(timeInMs > 5 ? "Had to wait to obtain the lock"
```

Comment:
Why 5?

Reply:
Arbitrary, I just saw locally that it was 0-2m each time. It's probably more on the server because of the network time to redis (locally it's all on one machine), so if I see a lot of "Had to wait to obtain the lock"s with 10-20 seconds, I'll up it. It's not super crucial, just wanted to give someone a cue they could find at a quick glance (or with Cmd-F), so if it takes some calibration/correction later, it's not a big deal.